main.cpp

```diff
@@ -227,24 +227,47 @@ class CDnsThread {
     dns_opt.port = opts->nPort;
     dns_opt.nRequests = 0;
     cache.clear();
-    cache.reserve(1000);
     cacheTime = 0;
     cacheHits = 0;
     dbQueries = 0;
     nIPv4 = 0;
     nIPv6 = 0;
-    cacheHit(true);
   }
 
   void run() {
     dnsserver(&dns_opt);
   }
 };
 
-extern "C" int GetIPList(void *data, addr_t* addr, int max, int ipv4, int ipv6) {
+extern "C" int GetIPList(void *data, char *requestedHostname, addr_t* addr, int max, int ipv4, int ipv6) {
   CDnsThread *thread = (CDnsThread*)data;
-  thread->cacheHit();
-  unsigned int size = thread->cache.size();
+
+  uint64_t requestedFlags = 0;
+  int hostlen = strlen(requestedHostname);
+  char maxHex[16+1]; //uint64_t as hex + null terminator
+  if (hostlen > 0 && requestedHostname[0] == 'x') {
+    for (unsigned int i = 1; i<hostlen; i++)
```

Comment:
> How would you then detect subdomains longer then 17 chars?

if (endptr > 17+ptr) {

> How would you then reject non hex chars?

strtoul stops at the first non-hex character. Just check whether endptr points to a '.'.

> How would you then reject leading zeros?

Check that the first character after 'x' is not '0'.